src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1329,11 +1317,15 @@ def _task_fetch_from_data_source(
     result = conn.cursor().execute(query).fetchall()
     columns = [col[0] for col in schema]
     df = pd.DataFrame.from_records(result, columns=columns)
+
+    # convert timestamp and date to string to avoid but in SQL
```

Comment:
better link the bug jira in the comment to give context on why we choose this way
```suggestion
    # convert timestamp and date to string to avoid bug in SQL
```

Reply:
will do